File: app/src/main/java/harkor/mycryptocurrency/DataRepository.java
package harkor.mycryptocurrency;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class DataRepository {
    private  static DataRepository instance;

    public static DataRepository getInstance(){
        if(instance==null){
            instance=new DataRepository();
        }
        return instance;
    }

    public MutableLiveData<String> getAmount(){
        final MutableLiveData<String> amount=new MutableLiveData<>();
        amount.setValue("$1000000");
        return amount;
    }


}

File: app/src/main/java/harkor/mycryptocurrency/DataRepository.kt
package harkor.mycryptocurrency

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Observer

class DataRepository {

    val amount: MutableLiveData<String>
        get() {
            val amount = MutableLiveData<String>()
            amount.value = "$1000000"
            return amount
        }
    val retrofit =RetrofitClientInstance.instance.create(CoinpaprikaApi::class.java)

    companion object {
        private var instance: DataRepository? = null

        fun getInstance(): DataRepository {
            if (instance == null) {
                instance = DataRepository()
            }
            return instance as DataRepository
        }
    }
    fun deachData(){
        //retrofit.getCoinsDataList().observe(, Observer {  })
    }


}
